sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryItem.cs

```diff
@@ -28,26 +20,18 @@ internal class TelemetryPartA
             [TelemetryType.Event] = "Event",
         };
 
-        internal static string RoleName { get; set; }
-
-        internal static string RoleInstance { get; set; }
-
-        internal static TelemetryItem GetTelemetryItem(Activity activity, ref TagEnumerationState monitorTags, Resource resource, string instrumentationKey)
+        internal static TelemetryItem GetTelemetry(Activity activity, ref TagEnumerationState monitorTags, string instrumentationKey)
         {
             TelemetryItem telemetryItem = new TelemetryItem(PartA_Name_Mapping[activity.GetTelemetryType()], FormatUtcTimestamp(activity.StartTimeUtc))
```

Comment:
Just an idea: what if we do this in the calling method itself?
And change this to `internal void Initialize(Activity activity, ref TagEnumerationState monitorTags)`

Reply:
We could also a constructor initialization, will take a look into it.